### DISCUSSION TITLE
Help (re-)defining fortran gcc linter on macOS

### DISCUSSION THREAD

**Discussion by mathomp4** [w0rp/ale#4820] [Q&A - Ask for help with problems]
All,

On Linux, using ALE on Fortran code works for me because `gcc` is GNU `gcc` and `:ALEInfo` shows me it is assuming that:
```
(executable check - success) gcc
(finished - exit code 1) ['/bin/bash', '-c', '''gcc'' -S -x f95 -fsyntax-only -ffree-form -Wall - < ''/gpfsm/dnb34/tdirs/login/discover32.102881.mathomp4/vZ0BcBh/3/nan_finder.F90''']

<<<OUTPUT STARTS>>>
<stdin>:130:117:

...
```

But on macOS `gcc` is `clang` and:
```
(executable check - success) gcc
(finished - exit code 1) ['/bin/bash', '-c', '''gcc'' -S -x f95 -fsyntax-only -ffree-form -Wall - < ''/var/folders/31/16x8_l1s72s2kw0w7byr03qm0000gp/T/vqgiof6/3/nan_finder.F90''']

<<<OUTPUT STARTS>>>
clang: error: unknown argument: '-ffree-form'
<<<OUTPUT ENDS>>>
```

And it looks like ALE says the only linters for fortran are:
```
 Current Filetype: fortran
Available Linters: ['gcc', 'language_server']
```

So I was wondering, is there a way to tell ALE that on this machine I want to use `gcc-14` say from Brew or even `/path/to/real/gcc`? Rather than assuming that `gcc` is GNU GCC?

**Comment by hsanson** [comment]
This configuration `g:ale_fortran_gcc_executable  ` allows to force a specific binary.  More details in `:h ale-fortran--options`.
